sdk/typescript/src/providers/json-rpc-provider.ts

```diff
@@ -748,4 +750,18 @@ export class JsonRpcProvider extends Provider {
   async unsubscribeEvent(id: SubscriptionId): Promise<boolean> {
     return this.wsClient.unsubscribeEvent(id);
   }
+
+  async dryRunTransaction(txBytes: string): Promise<TransactionEffects> {
```

Comment:
Is always supported from network/fullnode/env? Do we need to handle cases that it's not?